lib/matplotlib/cbook.py

```diff
@@ -902,14 +917,14 @@ def __iter__(self):
 
         The iterator is invalid if interleaved with calls to join().
         """
-        unique_groups = {id(group): group for group in self._mapping.values()}
+        unique_groups = {id(group): group for _, group in self._mapping.values()}
         for group in unique_groups.values():
-            yield [x for x in group]
+            yield sorted(group, key=self._mapping.__getitem__)
 
     def get_siblings(self, a):
         """Return all of the items joined with *a*, including itself."""
-        siblings = self._mapping.get(a, [a])
-        return [x for x in siblings]
+        _, siblings = self._mapping.get(a, (None, [a]))
+        return sorted(siblings, key=self._mapping.get)
```

Comment:
Why is one using `.__getitem__` and one `.get`?

Reply:
They're both using `.get`?

Reply:
I mean the sorting key.

Reply:
Lines 922 vs. 927?

Reply:
Ah, this is because `__iter__` only returns items already in the grouper (so `__getitem__` is guaranteed to succeed), whereas `get_siblings` can be called with something *not* in the Grouper, in which case `siblings = [a]` and the sort key will again fail if using a straight `__getitem__`.